orchestrator/careplancontributor/ehr/notifier.go

```diff
@@ -0,0 +1,373 @@
+package ehr
+
+import (
+	"encoding/json"
+	"fmt"
+	fhirclient "github.com/SanteonNL/go-fhir-client"
+	"github.com/SanteonNL/orca/orchestrator/lib/coolfhir"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+	"slices"
+	"strings"
+)
+
+type Notifier interface {
+	NotifyTaskAccepted(cpsClient fhirclient.Client, task *fhir.Task) error
+}
+
+type notifier struct {
+	kafkaClient KafkaClient
+}
+
+func NewNotifier(kafkaClient KafkaClient) Notifier {
+	return &notifier{kafkaClient}
+}
+
+// BundleSet represents a set of FHIR bundles associated with a task.
+type BundleSet struct {
+	Id      string
+	task    string
+	Bundles []fhir.Bundle `json:"bundles"`
+}
+
+// addBundle adds one or more FHIR bundles to the BundleSet.
+func (b *BundleSet) addBundle(bundle ...fhir.Bundle) {
+	b.Bundles = append(b.Bundles, bundle...)
+}
+
+// NotifyTaskAccepted handles the notification process when a task is accepted.
+// It fetches related FHIR resources and sends the data to Kafka.
+//
+// Parameters:
+//   - task: The FHIR task that was accepted.
+//
+// Returns:
+//   - error: An error if the notification process fails.
+func (n *notifier) NotifyTaskAccepted(cpsClient fhirclient.Client, task *fhir.Task) error {
+
+	ref := "Task/" + *task.Id
+	log.Info().Msgf("NotifyTaskAccepted Task (ref=%s)", ref)
```

Comment:
Maybe rephrase this something like "Notifying EHR of accepted Task over Kafka", since now it's very technical